inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -97,16 +97,16 @@ void InLaneCruisingPlugin::plan_trajectory_callback(
   original_trajectory.header.stamp = nh_->now();
   original_trajectory.trajectory_id = boost::uuids::to_string(boost::uuids::random_generator()());
 
-  original_trajectory.trajectory_points = basic_autonomy:: waypoint_generation::compose_lanefollow_trajectory_from_path(points_and_target_speeds, 
-                                                                                req->vehicle_state, req->header.stamp, wm_, ending_state_before_buffer_, debug_msg_, 
+  original_trajectory.trajectory_points = basic_autonomy:: waypoint_generation::compose_lanefollow_trajectory_from_path(points_and_target_speeds,
+                                                                                req->vehicle_state, req->header.stamp, wm_, ending_state_before_buffer_, debug_msg_,
                                                                                 wpg_detail_config); // Compute the trajectory
   original_trajectory.initial_longitudinal_velocity = std::max(req->vehicle_state.longitudinal_vel, config_.minimum_speed);
 
   // Set the planning plugin field name
   for (auto& p : original_trajectory.trajectory_points) {
     p.planner_plugin_name = plugin_name_;
   }
-  
+
   // Aside from the flag, ILC should not call yield_plugin on invalid trajectories
   if (config_.enable_object_avoidance && original_trajectory.trajectory_points.size() >= 2)
```

Comment:
the actual function is named `modify_trajectory_to_yield_to_obstacles` now in basic_autonomy (so that it can be used by other tactical plugins) and its inside if logics have also been refactored to follow your suggestion